### ISSUE TITLE
[Bug] `php artisan backpack:require:devtools` doesn't work

### ISSUE THREAD

**Issue by tabacitu** [Laravel-Backpack/CRUD#3886]
# Bug report

### What I did

In a Backpack demo, I ran `php artisan backpack:require:devtools` to install DevTools.

### What I expected to happen

Prompt for user & password and install DevTools.

### What happened

Did NOT prompt for user & password.

The output made me think it's installed now:

<img width="782" alt="Screenshot 2021-10-20 at 11 33 23" src="https://user-images.githubusercontent.com/1032474/138057860-0d291798-7e64-4b25-ac70-320ae486daee.png">

That last command was never run, though. I didn't notice it at the time, but `php artisan backpack:devtools:install` definitely wasn't run, since I didn't see the big fat notice of not having DevTools in production. Anyway, then I go to my browser expecting to be able to click the `DevTools` menu item, but here's what I see:

<img width="1440" alt="Screenshot 2021-10-20 at 11 38 13" src="https://user-images.githubusercontent.com/1032474/138058640-d301a6d5-4d40-44bc-8c94-ceb8bb2a1425.png">

Now my entire Laravel app is broken. Note that the only changes it ACTUALLY made were in `composer.json`:

<img width="1238" alt="Screenshot 2021-10-20 at 11 38 52" src="https://user-images.githubusercontent.com/1032474/138058741-17ecb5ce-06e1-4e86-a010-c17b91f809d2.png">

But apparently the process stopped after `composer config` for some reason, didn't do any of the next stuff, but the output makes it look like it DID, and it's really confusing.

Anyway, I could get the Laravel app to work again by running `composer install`. But... still no DevTools. Only a `composer update` would do that I guess. 

### What I've already tried to fix it

- Tried doing the same but with the `--debug` flag, didn't produce any extra output;
- Tried installing using the instructions on the DevTools README - worked fine;
- Tried running `php artisan backpack:install` - same problem (no surprise there);

### Versions

```
### PHP VERSION:
PHP 7.4.21 (cli) (built: Jul 12 2021 03:04:54) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Xdebug v3.0.4, Copyright (c) 2002-2021, by Derick Rethans
    with Zend OPcache v7.4.21, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
v8.62.0@60a7e00488167ce2babf3a2aeb3677e48aaf39be

### BACKPACK VERSION:
4.1.54@6f0ddffbe2017ef0e0d9becf5e9f422a81d94c45
```
-----

Can you reproduce this @promatik ?

**Comment by tabacitu** [comment]
At the very least, if the process fails we should tell user that we couldn't install automatically, and they should follow the manual process in https://backpackforlaravel.com/products/devtools (which we know does work).

**Comment by pxpm** [comment]
Unfortunatelly @tabacitu it's working here. 

![image](https://user-images.githubusercontent.com/7188159/139831057-10e3d07d-4e67-4917-abe5-2ba70053f47e.png)

Same project I tried before. 

Maybe we can save `$buffer` output by default in some log file? `devtools.log` so if/when something like this happen we can have some backtrace about it ? 

Let me know!
Pedro

**Comment by tabacitu** [comment]
Damn it 😅 I mean... that's good news, _it's working_, but... you know 😅

I'll close this. I agree, we should consider logging info for debugging this, if we ever have problems with it.

**Comment by pxpm** [comment]
@tabacitu exceptional news on this! 

I was working on the logging we talked, it seemd reasonable to atleast know why it fails. 

I tried again in another demo project I have here. I logged the error and found the culprit!

```php
[2021-11-02T11:38:38.568356+00:00] devtools_errors.ERROR: [37;41mYour requirements could not be resolved to an installable set of packages.[39;49m    Problem 1     - backpack/devtools 1.0.14 requires laravel-shift/blueprint >=1.25 -> satisfiable by laravel-shift/blueprint[v1.25.0, v2.0.0].     - backpack/devtools[1.0.10, ..., 1.0.13] require laravel-shift/blueprint dev-master -> satisfiable by laravel-shift/blueprint[dev-master].     - backpack/devtools[1.0.0, ..., 1.0.9] require laravel-shift/blueprint ^1.23.2 -> satisfiable by laravel-shift/blueprint[v1.23.2, ..., v1.25.0].     - backpack/devtools[1.0.15, ..., 1.0.28] require laravel-shift/blueprint ^1.25 -> satisfiable by laravel-shift/blueprint[v1.25.0].     - laravel-shift/blueprint[dev-master, v1.24.2, ..., v1.25.0, v2.0.0] require doctrine/dbal ^2.9 || ^3.0 -> found doctrine/dbal[v2.9.0, ..., 2.13.x-dev, 3.0.0, ..., 3.2.x-dev] but it conflicts with your root composer.json require (2.6).     - laravel-shift/blueprint[v1.23.2, ..., v1.24.1] require doctrine/dbal ^2.9|^3.0 -> found doctrine/dbal[v2.9.0, ..., 2.13.x-dev, 3.0.0, ..., 3.2.x-dev] but it conflicts with your root composer.json require (2.6).     - Root composer.json requires backpack/devtools ^1.0 -> satisfiable by backpack/devtools[1.0.0, ..., 1.0.28].   [37;41mInstallation failed, reverting ./composer.json and ./composer.lock to their original content.[39;49m  [] []
```

It seems the root cause for the failing process is the fact that we require `doctrine/dbal 2.6` in demo project to support older versions of Laravel, and in Devtools we don't support those versions and require an higher version. 

I've tested upgrading `dbal` to `3.*` and after run the command without issues. Can you try it too @tabacitu ?

Best,
Pedro

**Comment by tabacitu** [comment]
Awesome - good job @pxpm ! Now that we know what's happening, @promatik should be able to replicate it - so I'll assign him. 

Where I think we should focus here is:

1. Not fail silently - the dev to know what went wrong and what they can do about it (at least point to the manual install).
2. If it fails, that it does NOT completely break the app 😅

**Comment by tabacitu** [comment]
Also - just wondering here. Wouldn't this be fixed _automatically_ by composer if we installed DevTools using `composer require --dev --with-all-dependencies backpack/devtools` instead? Would that be to invasive?

**Comment by promatik** [comment]
@tabacitu we already do that 😅
But in this case the it is really impossible to install/update, DevTools depends on blueprint >1.25, and blueprint 1.25 depends on dbal 3. Composer can't handle that 🤷‍♂️

In the between, I added this error to the install log.

![image](https://user-images.githubusercontent.com/1838187/140663401-9bdbf843-4d36-415a-96d3-f16cf5b15dd9.png)

The fix is now on #3843.

I'll close this one for now ✌
